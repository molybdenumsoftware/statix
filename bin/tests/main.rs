mod util {
    #[macro_export]
    macro_rules! test_lint {
        ($tname:ident, $($tail:tt)*) => {
            test_lint!($tname);
            test_lint!($($tail)*);
        };
        ($tname:ident, $($tail:tt)*) => {
                test_lint!($tname);
                test_lint!($($tail)*);
        };
        ($tname:ident) => {
            paste::paste! {
                #[test]
                fn [<$tname _lint>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_lint"), file_path, &["check"]);
                }

                #[test]
                fn [<$tname _fix>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_fix"), file_path, &["fix", "--dry-run"]);
                }
            }

        };
    }
}

fn test_cli(test_name: &str, file_path: &str, args: &[&str]) {
    let output = std::process::Command::new("cargo")
        .arg("run")
        .arg("--")
        .args(args)
        .arg(file_path)
        .output()
        .expect("command runs successfully");

    let stdout = strip_ansi_escapes::strip(output.stdout).unwrap();
    let stdout = String::from_utf8(stdout).unwrap();

    insta::assert_snapshot!(test_name, &stdout);
}

test_lint! {
    collapsible_let_in,
    eta_reduction,
    useless_parens,
    empty_pattern,
    redundant_pattern_bind,
    unquoted_uri,
    empty_inherit,
    deprecated_to_path,
    useless_has_attr,
    repeated_keys,
    empty_list_concat
}
