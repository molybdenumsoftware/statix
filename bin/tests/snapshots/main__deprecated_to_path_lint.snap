---
source: bin/tests/main.rs
expression: "& out"
---
[W17] Warning: Found usage of deprecated builtin toPath
   ╭─[data/deprecated_to_path.nix:2:4]
   │
 2 │   (builtins.toPath x)
   ·    ────────┬────────  
   ·            ╰────────── builtins.toPath is deprecated, see :doc builtins.toPath within the REPL for more
───╯
[W17] Warning: Found usage of deprecated builtin toPath
   ╭─[data/deprecated_to_path.nix:3:4]
   │
 3 │   (toPath x)
   ·    ────┬───  
   ·        ╰───── toPath is deprecated, see :doc builtins.toPath within the REPL for more
───╯
[W17] Warning: Found usage of deprecated builtin toPath
   ╭─[data/deprecated_to_path.nix:4:4]
   │
 4 │   (toPath "/abc/def")
   ·    ────────┬────────  
   ·            ╰────────── toPath is deprecated, see :doc builtins.toPath within the REPL for more
───╯
[W17] Warning: Found usage of deprecated builtin toPath
   ╭─[data/deprecated_to_path.nix:5:4]
   │
 5 │   (builtins.toPath "/some/path")
   ·    ──────────────┬─────────────  
   ·                  ╰─────────────── builtins.toPath is deprecated, see :doc builtins.toPath within the REPL for more
───╯
